### PR TITLE
Make the test-history index color failures based on frequency.

### PR DESCRIPTION
**Pull request by rmmh**
Colors are red/orange/yellowish (chosen to be easily distinguishable).

image

### CODE REVIEW

File: jenkins/test-history/gen_html.py [hunk 1]
```diff
@@ -57,6 +57,28 @@
...
+    extensions=['jinja2.ext.autoescape'],
+    trim_blocks=True,
+    autoescape=True)
+
+
+def failure_class(passed, failed):
+    if failed == 0:
+        return ''
+    elif passed == 0:
+        return ' job-broken'
```
Comment (spxtr) [inline jenkins/test-history/gen_html.py]: This is a smell that we should be keeping a list and joining it with a space.

Comment (rmmh) [inline jenkins/test-history/gen_html.py]: oh, this is actually unnecessary. jinja is smart enough to trim the attribute for me!

File: jenkins/test-history/gen_html.py [hunk 2]
```diff
@@ -222,23 +244,21 @@ def main(in_path, buckets_path, out_dir):
...
     blocking_job_summaries = filter(lambda s: s.name in BLOCKING_JOBS, summaries)
 
-    with open('{}/index.html'.format(templates_path)) as index_template_file:
-        index_template = jinja2.Template(index_template_file.read())
+    index_template = JINJA_ENV.get_template('index.html')
     index_html = index_template.render({
         'last_updated': time.strftime('%a %b %d %T %Z'),
         'job_groups': [blocking_job_summaries, summaries],
-        'bad_tests': sorted(bad_tests.values(), key=lambda t: t['failed'], reverse=True),
+        'bad_tests': sorted(bad_tests.values(), key=lambda t: t['failed'] / (t['passed'] + .01), reverse=True),
```
Comment (spxtr) [inline jenkins/test-history/gen_html.py]: `failed / (passed + failed)`

Comment (rmmh) [inline jenkins/test-history/gen_html.py]: Done. bad_tests always has nonzero failed.

**Comment by spxtr** [comment]
LGTM

**Comment by jlowdermilk** [comment]
Excellent!

**Pull request merged**